Xamarin.Forms.ControlGallery.iOS/Tests/NavigationTests.cs

```diff
@@ -14,12 +14,15 @@ public void NavigationRendererDoubleDisposal()
 				Title = "root",
 				Content = new Label { Text = "Hello" }
 			};
-			var navPage = new NavigationPage(root);
-			var renderer = GetRenderer(navPage);
 
-			// Calling Dispose more than once should be fine
-			renderer.Dispose();
-			renderer.Dispose();
+			Device.InvokeOnMainThreadAsync(() => { 
+				var navPage = new NavigationPage(root);
+				var renderer = GetRenderer(navPage);
+
+				// Calling Dispose more than once should be fine
+				renderer.Dispose();
```

Comment:
Could you please leave a comment why do we call Dispose twice

Reply:
I think it's because dispose method should handle multiple calls and not cause a crash if called after the object has already been disposed.
So a dispose test should also test for this case.

Reply:
We don't really do it on purpose, but `Dispose(true)` can occasionally be called more than once in some race conditions when we're doing things like popping/replacing pages, when we've got weak event references, and possibly when we've just written bad code. 
And `Dispose(false)` might happen under some conditions when objects are being finalized. 

So it's just good practice to ensure your `Dispose` implementations can gracefully handle being called more than once.